### DISCUSSION TITLE
NA's in dca net benefit output for survival model

### DISCUSSION THREAD

**Discussion by Meech81** [MSKCC-Epi-Bio/decisioncurveanalysis#2] [Q&A] [answered]
I’m currently using the dcurves package to create decision curves in R (version 4.2.2) for some Cox PH models.

I’m finding that when I run the dca function, I’m getting chunks of missing (blank cell) values for tp_rate, fp_rate and net_benefit, followed by a section where net benefit is correctly calculated as 0. As far as I can work out, there's no reason why the net benefit isn't calculable in these regions. 

The following example is from my data, I've only included 10 obvs for ease to demonstrate, my actual data has 455 obvs. The range of thresholds affected are different for each of my several models.

Any help would be much appreciated.

Thanks

```
df <- data.frame(
time = c(5.428571, 144.000000, 246.000000,  47.142857,  52.285714,  32.428571,  16.000000,  39.285714, 207.000000,53.000000),
event = c(1, 1, 0, 1, 1, 1, 1, 1, 0, 1),
pred = c(0.25832215, 0.19859299, 0.12332482, 0.07242517, 0.58521249, 0.16277858, 0.55437640, 0.30077087, 0.10966112, 0.50300771))

dca_object <- dca(Surv(time, event) ~ pred, 
data = df,
time = 53.2)

dca_df <- dca_object$dca
as.data.frame(dca_df[dca_df$label=="pred",])
```

**Comment by ddsjoberg** [comment, marked as answer]
@Meech81 the reason you see NA is because the net benefit is not calculable at those thresholds. (Details in this file [dca.zip](https://github.com/MSKCC-Epi-Bio/decisioncurveanalysis/files/11592318/dca.zip))

In the survival or time-to-event setting, here's how we calculate the net benefit.
![image](https://github.com/MSKCC-Epi-Bio/decisioncurveanalysis/assets/26774684/6bf45f4b-425b-461d-91d3-2658c49489f0)
where
![image](https://github.com/MSKCC-Epi-Bio/decisioncurveanalysis/assets/26774684/5d785aa0-e371-4b2f-946c-0a0770969273)

In your example you see NA above the 20% threshold, because the probability of the event is not calculable as there are no subjects with observed follow-up at or later than time 53.2. 

I suggest you focus on a time point where you have more data if that makes sense clinically, and focus on the range of threshold probabilities are actually being made. For example, if the model will only be used around 5%-10%, only show those thresholds on the figure.

**Reply by Meech81** [reply]
@ddsjoberg thank you so much, that makes perfect sense. 

The only thing that confuses me is why we start to see net benefit = 0 (rather than NA) from threshold = 0.59 onwards?

**Reply by ddsjoberg** [reply]
The reason for this (and perhaps could be updated) is that before any Bayes rules are applied, I check if there are any obs above the threshold. If there are none, then I assume the TP rate is 0 and the FP rate is 0 because these are no "test-positives". This makes perfect sense in the binary setting, but there is some nuance in the survival setting where we could also then check if there are observations at/above the timepoint selected.

**Reply by Meech81** [reply]
Thanks @ddsjoberg

**Reply by Meech81** [reply]
@ddsjoberg thanks again for your help on this. I have one more question – would you say that it’s reasonable to apply the same logic (i.e. assuming the TP and FP rates are 0) to the regions where the dca function calculates NA's and assign those values as zero?

For context-I'm calculating the optimism (via 200 bootstraps) around the net benefit for my models. Within the bootstraps, I’m left with several regions of NA’s for net benefit 'orig' (net benefit from the bootstrapped model applied to the original data) but have values for the same regions for net benefit 'boot' (net benefit from the bootstrapped model applied to the bootstrapped data) and vice versa. When I’m subtracting the two, I’m therefore getting NA’s essentially excluding those regions from my estimate of optimism. Given what you’ve said above, I think that it would be reasonable to assign the NA’s as zero, therefore allowing those regions to be included.

I would be very grateful for your thoughts on this.

**Reply by ddsjoberg** [reply]
I don't think I agree with that logic.

![image](https://github.com/MSKCC-Epi-Bio/decisioncurveanalysis/assets/26774684/25b21d39-9b19-4c57-b4ad-7155b9fe926c)

In this probabilities, Pr(T+) is always calculable. In the case where there are no observations with observed risk greater the threshold probability, Pr(T+) will be zero. Any number multiplied by zero is zero, without even calculating the conditional probability we know the result is zero.

The conditional probability where you see NA is simply saying, "We don't have enough data to estimate this probability." In this case, we have an unknown number (not estimable with our data) multiplied by a non-zero Pr(T+)....we can't assume that this quantity is zero.

**Reply by Meech81** [reply]
Thanks @ddsjoberg.

So, in fact, should the zero's for the regions after the NA's also be NA?

**Reply by ddsjoberg** [reply]
We know the net benefit for those thresholds is zero, so I wouldn't agree 100% with that statement. Zero multiplied by any real number (even if we don't have the data to estimate that number) is zero.

**Reply by ddsjoberg** [reply]
That said, I would reduce the range of the DCA curve you're showing in the paper to only show thresholds where you can get a reasonable estimate of the net benefit and a range of thresholds where decisions are actually being made. If this area where we're seeing NAs and 0s is in that decision threshold, the the model isn't good enough to help with those decisions.

**Reply by Meech81** [reply]
@ddsjoberg thanks for all your help on this, it's much appreciated.
